cpp/autosar/src/rules/A7-1-2/VariableMissingConstexpr.ql

```diff
@@ -33,6 +33,15 @@ predicate isTypeZeroInitializable(Type t) {
   t.getUnderlyingType() instanceof ArrayType
 }
 
+predicate isCompileTimeEvaluated(Call call) {
+  call.getTarget().isConstexpr() and
+  forall(Expr arg | arg = call.getAnArgument() |
```

Comment:
I think this logic needs another consideration

literals flowing into a var might not be sufficient

consider:
```
int a = 0;
  if (p> 0){
   a= 1;  
  }else{
    a =p+1; 
  }
    
  int x = add(a, 2); // cannot be made constexpr but currently being reported
```

if the literal comes from a constexpr var itself though then it is fine - like the following (actually also fine if a is not declared `constexpr` - but in this case it also could be):
```
  constexpr int a = 0;    
  int x = add(a, 2); // NON_COMPLIANT
```

Reply:
Good catch. I definitely need to consider more corner cases.